samigo/samigo-hibernate/src/java/org/sakaiproject/tool/assessment/data/dao/assessment/ExtendedTime.java

```diff
@@ -79,8 +79,8 @@ public boolean  equals              (final Object obj) {
         if(obj instanceof ExtendedTime) {
             final ExtendedTime other = (ExtendedTime) obj;
             return new EqualsBuilder()
-                    .append(assessment.getAssessmentBaseId(), other.assessment.getAssessmentBaseId())
-                    .append(pubAssessment.getPublishedAssessmentId(), other.getPubAssessmentId())
+                    .append(assessment == null ? null : assessment.getAssessmentBaseId(), other.assessment.getAssessmentBaseId())
```

Comment:
If other.assessment == null, this will cause problems. Using other.getAssessmentId() is null safe.

You could also use getAssessmentId() and getPubAssessmentId() instead of the ? construct, if you so desired.